fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/OCFLPersistentStorageUtils.java

```diff
@@ -170,21 +175,58 @@ public static RdfStream getRdfStream(final String identifier,
         }
     }
 
-    private static  String resolveVersionId(final OCFLObjectSession objSession, final Instant version) {
-        //TODO Implement resolution of a version id (OCFL-speak) from an instant (memento-speak)
-       return null;
+    private static String resolveVersionId(final OCFLObjectSession objSession, final Instant version)
+            throws PersistentStorageException {
+        if (version != null) {
+            return objSession.listVersions()
+                    .stream()
+                    .filter(vd -> {
+                        //filter by comparing Epoch seconds since
+                        //Memento has second granularity while OCFL versions are
+                        //millisecond granularity
+                        return vd.getCreated()
+                                .toInstant()
+                                .toEpochMilli() / 1000 == version.toEpochMilli() / 1000;
+                    }).map(vd -> vd.getVersionId().toString()) //return the versionId the matches
+                    .findFirst()
+                    .orElseThrow(() -> {
+                        //otherwise throw an exception.
+                        return new PersistentItemNotFoundException(format(
+                                "There is no version in %s with a created date matchin %s",
+                                objSession, version));
+                    });
+        } else {
+            //return null if the instant is null
+            return null;
+        }
+    }
+
+    /**
+     * A utility method that returns a list of {@link java.time.Instant} objects representing versions
+     * accessible from the OCFL Object represented by the session.
+     *
+     * @param objSession The OCFL object session
+     * @return A list of Instant objects
+     * @throws PersistentStorageException
```

Comment:
Needs description

Reply:
will do.

Reply:
Not sure what's needed here.  But I will make the immutability of the versions explicit.

Reply:
The `@throws` in many of these does not have a description as to why the Exception is thrown. This causes a Warning message to be output when compiling.